File: slf4j/src/main/java/org/slf4j/impl/StaticLoggerBinder.java
package org.slf4j.impl;

import com.scribe.slf4j.ScribeLoggerFactory;
import org.slf4j.ILoggerFactory;
import org.slf4j.spi.LoggerFactoryBinder;

public class StaticLoggerBinder implements LoggerFactoryBinder {
    private static final StaticLoggerBinder SINGLETON = new StaticLoggerBinder();
    private static final String loggerFactoryClassStr = ScribeLoggerFactory.class.getName();

    public static String REQUESTED_API_VERSION = "1.7.15";

    private final ILoggerFactory loggerFactory;

    private StaticLoggerBinder() {
        loggerFactory = new ScribeLoggerFactory();
    }

    public static StaticLoggerBinder getSingleton() {
        return SINGLETON;
    }

    @Override
    public ILoggerFactory getLoggerFactory() {
        return loggerFactory;
    }

    @Override
    public String getLoggerFactoryClassStr() {
        return loggerFactoryClassStr;
    }
}

File: slf4j/src/main/scala/org/slf4j/impl/StaticLoggerBinder.scala
package org.slf4j.impl

import scribe.slf4j.ScribeLoggerFactory
import org.slf4j.ILoggerFactory
import org.slf4j.spi.LoggerFactoryBinder

class StaticLoggerBinder private() extends LoggerFactoryBinder {
  private val factory = new ScribeLoggerFactory
  private val classString = classOf[ScribeLoggerFactory].getName

  override def getLoggerFactory: ILoggerFactory = factory

  override def getLoggerFactoryClassStr: String = classString
}

object StaticLoggerBinder extends StaticLoggerBinder {
  val REQUESTED_API_VERSION = "1.7.15"

  def getSingleton: StaticLoggerBinder = this
}